tensorflow/compiler/xla/service/gpu/gemm_thunk.cc

```diff
@@ -124,7 +124,15 @@ static bool DoGemmWithAlgorithm(
                                             : se::blas::Transpose::kNoTranspose;
   auto k = lhs_matrix.transpose ? lhs_matrix.num_rows : lhs_matrix.num_cols;
 
-  if (algorithm) {
+  // Ignore the "algorithm" field on the ROCm platform. This is because
+  // autotuning for GEMM is not yet available on the ROCm platform
```

Comment:
Can we fix the unit test instead?

Reply:
@sanjoy 

I thought about going that route, but decided against it for the following reasons.
* it requires changing the unit test to not use the `algorithm` attribute (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/tests/mlir_gemm_test.cc#L34 ). This will apply to the CUDA side too, and result in that path not being tested on the CUDA side too.
* the same thing could happen again, if/when more tests are added that explicitly set the `algorithm` attribute in a gemm call. Then we would need to go this whole process again (detect it, file a PR to fix it, etc)

If changing the unit-test is more preferable, then I can ammend the PR to make that change. Let me know.

Reply:
@sanjoy gentle ping

Reply:
Actually instead of doing that, WDYT about changing the ROCm-BLAS implementation to simply forward the call from ThenBlasGemmWithAlgo to ThenBlasGemm?

Reply:
Thanks Deven, keeping this check as is or George's suggestion make sense to me then.

However, I'm not sure if this check is sound -- it just checks if `ROCM` is linked in right, not that we are compiling for ROCm?  I know that for TensorFlow they are equivalent, but e.g. for JAX it might not be.

Reply:
@cheshire I looked into your suggestion, and I would like to implement it as a separate PR.  even though ROCm only supports 1 algorithm choice right now, we do have all the hooks in place to be able to properly implement support for DoBlasGemmWithAlgorithm

@sanjoy I updated the implementation to make the ROCm exclusion a compile time decision

please re-review

Reply:
@cheshire @sanjoy gentle ping

Reply:
I'm still not sure I like that approach, sorry. I'm trying to minimize the places where we have `#ifdef`, having it here seems like the wrong level of abstraction. Fixing unit test indeed seems preferable, since it's wrong to select an algorithm for the ROCm platform.

Reply:
> . Fixing unit test indeed seems preferable, since it's wrong to select an algorithm for the ROCm platform.

done...please re-review